File: tests/machine/x/ocaml/dataset_sort_take_limit.ml
let rec __show v =
  let open Obj in
  let rec list_aux o =
    if is_int o && (magic (obj o) : int) = 0 then "" else
     let hd = field o 0 in
     let tl = field o 1 in
     let rest = list_aux tl in
     if rest = "" then __show (obj hd) else __show (obj hd) ^ "; " ^ rest
  in
  let r = repr v in
  if is_int r then string_of_int (magic v) else
  match tag r with
    | 0 -> if size r = 0 then "[]" else "[" ^ list_aux r ^ "]"
    | 252 -> (magic v : string)
    | 253 -> string_of_float (magic v)
    | _ -> "<value>"

exception Break
exception Continue


let products = [[("name",Obj.repr ("Laptop"));("price",Obj.repr (1500))];[("name",Obj.repr ("Smartphone"));("price",Obj.repr (900))];[("name",Obj.repr ("Tablet"));("price",Obj.repr (600))];[("name",Obj.repr ("Monitor"));("price",Obj.repr (300))];[("name",Obj.repr ("Keyboard"));("price",Obj.repr (100))];[("name",Obj.repr ("Mouse"));("price",Obj.repr (50))];[("name",Obj.repr ("Headphones"));("price",Obj.repr (200))]]
let expensive = (let __res0 = ref [] in
  List.iter (fun p ->
      __res0 := p :: !__res0;
  ) products;
List.rev !__res0)


let () =
  print_endline (__show ("--- Top products (excluding most expensive) ---"));
  let rec __loop1 lst =
    match lst with
      | [] -> ()
      | item::rest ->
        try
          print_endline (__show (Obj.obj (List.assoc "name" item)) ^ " " ^ __show ("costs $") ^ " " ^ __show (Obj.obj (List.assoc "price" item)));
        with Continue -> ()
        ; __loop1 rest
    in
    try __loop1 expensive with Break -> ()

File: tests/machine/x/java/dataset_sort_take_limit.java
import java.util.*;
class NamePrice {
	String name;
	int price;
	NamePrice(String name, int price) {
		this.name = name;
		this.price = price;
	}
	int size() { return 2; }
}
public class DatasetSortTakeLimit {
	public static void main(String[] args) {
	List<NamePrice> products = new ArrayList<>(Arrays.asList(new NamePrice("Laptop", 1500), new NamePrice("Smartphone", 900), new NamePrice("Tablet", 600), new NamePrice("Monitor", 300), new NamePrice("Keyboard", 100), new NamePrice("Mouse", 50), new NamePrice("Headphones", 200)));
	List<NamePrice> expensive = (new java.util.function.Supplier<List<NamePrice>>(){public List<NamePrice> get(){
	List<NamePrice> _res0 = new ArrayList<>();
	for (var p : products) {
		_res0.add(p);
	}
	return _res0;
}}).get();
	System.out.println("--- Top products (excluding most expensive) ---");
	for (NamePrice item : expensive) {
		System.out.println(item.name + " " + "costs $" + " " + item.price);
	}
	}
}
